open_rarity/resolver/opensea_api_helpers.py

```diff
@@ -177,6 +241,71 @@ def get_collection_with_metadata(
     return collection_with_metadata
 
 
+def get_collection_from_opensea(
+    opensea_collection_slug: str,
+) -> Collection:
+    """Fetches collection and token data with OpenSea endpoint and API key
+    and stores it in the Collection object
+
+    Parameters
+    ----------
+    opensea_collection_slug : str
+        collection slug on opensea's system
+
+    Returns
+    -------
+    Collection
+        collection abstraction
+
+    """
+    # Fetch collection metadata
+    collection_obj = fetch_opensea_collection_data(
+        slug=opensea_collection_slug
+    )
+    contracts = collection_obj["primary_asset_contracts"]
+    interfaces = set([contract["schema_name"] for contract in contracts])
+    stats = collection_obj["stats"]
+    if not interfaces.issubset(set(["ERC721", "ERC1155"])):
+        raise ValueError(
+            "We currently do not support non EVM standards at the moment"
+        )
+
+    total_supply = int(stats["total_supply"])
+
+    # Fetch token metadata
+    tokens: list[Token] = []
+    batch_size = 30
+    num_batches = math.ceil(total_supply / batch_size)
+    initial_token_id = 0
+
+    # Returns a list of `batch_size` token IDs, such that no token ID
+    # can exceed `max_token_id` (in which case len(return_value) < `batch_size`)
+    def get_token_ids(
+        batch_id: int, max_token_id: int = total_supply - 1
+    ) -> list[int]:
+        token_id_start = initial_token_id + (batch_id * batch_size)
+        token_id_end = int(min(token_id_start + batch_size - 1, max_token_id))
+        return [
+            token_id for token_id in range(token_id_start, token_id_end + 1)
+        ]
```

Comment:
`list(range(token_id_start, token_id_end + 1))` is more concise and twice as fast just fyi.